keel-api/src/main/kotlin/com/netflix/spinnaker/keel/api/plugins/ArtifactSupplier.kt

```diff
@@ -88,6 +88,11 @@ interface ArtifactSupplier<A : DeliveryArtifact, V : VersioningStrategy> : Spinn
    * This function is currently expected to make calls to CI systems.
    */
   suspend fun getArtifactMetadata(artifact: PublishedArtifact): ArtifactMetadata?
+
+  /**
+   * Given an artifact definition [DeliveryArtifact], and an actual artifact version as [PublishedArtifact], return whether this artifact should be processed and saved
+   */
+  fun shouldProcessArtifact(artifactDefinition: DeliveryArtifact, artifact: PublishedArtifact): Boolean
```

Comment:
I think for this we just need the published artifact.

The rules for processing should depend only on the type, and not someone's definition.

The rules we have are: 
- we ignore docker tags matching "latest"
- we ignore locally published debians, which we can tell by parsing the version and then trying to get the build number (if it is "local", we should ignore it)